### PR TITLE
test/test-functions: drop all prefixes

### PR DESCRIPTION
**Pull request by keszybz**
When parsing and installing binaries mentioned in Exec*= lines the
5ed0dcf4d552271115d96d8d22b1a25494b85277 commit added parsing logic to drop
prefixes, including handling duplicate exclamation marks. But this did not
handle arbitrary combination of multiple prefixes, ie. StartExec=+-/bin/sh was
parsed as -/bin/sh which then would fail to install.

Instead of using egrep and shell replacements, replace both with sed command
that does it all. This sed script extract a group of characters starting with a
/ up to the first space (if any) after the equals sign. This correctly handles
existing non-prefixed, prefixed, multiple-prefixed commands.

About half commands seem to repeat themself, thus sort -u cuts the list of
binaries to install about in half.

To validate change of behaviour both old and new functions were modified to
echo parsed binaries into separate files, and then diffed. The incorrect
-/bin/sh was missing in the new output.

Without this patch tests fail on default Ubuntu installs.

Resubmission of #7264.

### CODE REVIEW

File: test/test-functions [hunk 1]
```diff
@@ -425,9 +425,8 @@ install_execs() {
     export PKG_CONFIG_PATH=$BUILD_DIR/src/core/
     systemdsystemunitdir=$(pkg-config --variable=systemdsystemunitdir systemd)
     systemduserunitdir=$(pkg-config --variable=systemduserunitdir systemd)
-    egrep -ho '^Exec[^ ]*=[^ ]+' $initdir/{$systemdsystemunitdir,$systemduserunitdir}/*.service \
-         | while read i; do
-         i=${i##Exec*=}; i=${i##[@+\!-]}; i=${i##\!}
+    sed -n 's|^Exec[a-zA-Z]*=[^/]*\(/[^ ]*\).*|\1|gp' $initdir/{$systemdsystemunitdir,$systemduserunitdir}/*.service \
```
Comment (yuwata) [inline test/test-functions]: I could not find why it did not work when this was proposed.
But now, `ExecStart=` or friends does not request full path. So, I guess this does not work, e.g., `ExecStart=sh -x -c 'echo hoge'`.

File: test/test-functions [hunk 2]
```diff
@@ -504,9 +504,8 @@ install_execs() {
     export PKG_CONFIG_PATH=$BUILD_DIR/src/core/
     systemdsystemunitdir=$(pkg-config --variable=systemdsystemunitdir systemd)
     systemduserunitdir=$(pkg-config --variable=systemduserunitdir systemd)
-    egrep -ho '^Exec[^ ]*=[^ ]+' $initdir/{$systemdsystemunitdir,$systemduserunitdir}/*.service \
-         | while read i; do
-         i=${i##Exec*=}; i=${i##[@+\!-]}; i=${i##\!}
+    sed -r -n 's|^Exec[a-zA-Z]*=[@+!-]*([^ ]+).*|\1|gp' $initdir/{$systemdsystemunitdir,$systemduserunitdir}/*.service \
```
Comment (yuwata) [inline test/test-functions:507]: Not tested, but white spaces may need to be dropped. E.g.,
`  ExecStart    =    !    true`

Comment (keszybz) [reply test/test-functions:507]: Do we have any of those in tree? This is just for tests, so we don't need to make it completely general.

Comment (evverx) [reply test/test-functions:507]: I'm not sure all the prefixes should be dropped silently. In theory, the script should be clever enough to understand that getting "ExecStartPre=-/bin/plymouth" (as in `rescue.service) it shouldn't fail if "plymouth" can't be installed. Though, maybe it's just another bug which hasn't been discovered yet and should probably be fixed in a separate PR.

**Comment by keszybz** [comment]
@xnox ?

**Comment by xnox** [comment]
@keszybz @evverx the ubuntu CI runs in fairly minimal Ubuntu Cloud Image / minimal server install. However, my developer desktop is a regular Ubuntu Desktop installation with graphical stack. Ubuntu Desktop has more units available by default, some of which have multiple prefixes. Running these tests thus fails on my Ubuntu Desktop whilst doing systemd development. It would be really nice to finally merge this fix, as I constantly cherrypick my patch from ubuntu distro-packaging to apply in the local tree....

**Comment by evverx** [comment]
@xnox  I was just thought that the underlying issue had something to do with `ExecStartPre=-/bin/plymouth` I mentioned above. If it were the case than this PR wouldn't fix that. Anyway, I have no problem with the PR. Other issues could be fixed later.

**Comment by xnox** [comment]
At the time, I recall things like:
```
/lib/systemd/system/systemd-networkd.service:ExecStart=!!/lib/systemd/systemd-networkd
/lib/systemd/system/resolvconf-pull-resolved.service:ExecStart=+-/bin/sh -c 'cat /run/systemd/resolve/stub-resolv.conf | /sbin/resolvconf -a systemd-resolved'
```

causing trouble. I will open a new issue if things still don't work as expected.

**Pull request merged**